modP2pImpl/src/org/aion/p2p/impl/comm/NodeMgr.java

```diff
@@ -239,17 +239,20 @@ public void timeoutCheck() {
     }
 
     private void timeoutOutBound() {
-        Iterator<Integer> outboundIt = getOutboundNodes().keySet().iterator();
-        while (outboundIt.hasNext()) {
-            int outBound = outboundIt.next();
-            INode node = getOutboundNodes().get(outBound);
-            if (System.currentTimeMillis() - node.getTimestamp()
-                > TIMEOUT_OUTBOUND_NODES) {
-                p2pMgr.closeSocket(
-                    node.getChannel(),
-                    "outbound-timeout node=" + node.getIdShort() + " ip=" + node.getIpStr());
-                outboundIt.remove();
+        try {
+            Iterator<Map.Entry<Integer, INode>> it = outboundNodes.entrySet().iterator();
+            while (it.hasNext()) {
+                Map.Entry<Integer, INode> entry = it.next();
+                if (System.currentTimeMillis() - entry.getValue().getTimestamp()
+                    > TIMEOUT_OUTBOUND_NODES) {
+                    p2pMgr.closeSocket(entry.getValue().getChannel(),
+                        "outbound-timeout ip=" + entry.getValue().getIpStr());
+                    it.remove();
+                }
             }
+        } catch (Throwable e) {
```

Comment:
Is it possible to catch something more specific?  Throwable catches even things like OutOfMemoryError

Reply:
revise in
https://github.com/aionnetwork/aion/pull/539/commits/8ef1ace73afb32e34df31e45e46f033f63bb680b